### ISSUE TITLE
SwiftShader: Segfault

### ISSUE THREAD

**Issue by Jack-Clark** [mc-imperial/spirv-control-flow#9]
@vili-1 Can you confirm that a loop header can be its own continue target? See the CFG below.

Executing `amber -t spv1.3 -v 1.1 reduced-segfault.amber` leads to a segfault. I think this may be related to #8 as both seem to rely on there being input variable accesses in a selection header followed by a loop header.

I have attached [an archive](https://github.com/mc-imperial/spirv-control-flow/files/8452298/segfault.tar.gz) containing the files.

![Segfault](https://user-images.githubusercontent.com/13239506/162447348-1023cc17-6ff1-436f-ba52-ce76b749b6cf.png)

**Comment by vili-1** [comment]
Yes @Jack-Clark , a loop header can be its own continue target (this example is allowed - SHO is Entry). About the segfault I have to dig a bit into this.

**Comment by vili-1** [comment]
@Jack-Clark, I don't get segfault in my Mac

`Summary: 1 pass, 0 fail`

**Comment by Jack-Clark** [comment]
Great, thanks for confirming this. Regarding the segfault, dId you use SwiftShader or moltenVK?

**Comment by vili-1** [comment]
MoltenVK

```
$ ./amber -d -V
Amber        : d8acae6
SPIRV-Tools  : 97f1d485
SPIRV-Headers: dc77030
GLSLang      : 07a55839
Shaderc      : 821d564

Physical device properties:
  apiVersion: 1.1.170
  driverVersion: 10102
  vendorID: 32902
  deviceID: 35411
  deviceType: integrated gpu
  deviceName: Intel(R) Iris(TM) Plus Graphics
  driverName: MoltenVK
  driverInfo: 1.1.2
End of physical device properties.
```

**Comment by vili-1** [comment]
As stated in #8, we get an error when the node in the fleshing path is both a switch case and merge block of the switch. The successful execution that I mentioned above was using a fleshing path other than this. Maybe we have a chat with @afd at some point to clarify this (not clear to me at the moment).

**Comment by vili-1** [comment]
Loading the SwiftShader I get no error!

**Comment by Jack-Clark** [comment]
Hmmm, which commit are you using for SwiftShader?
